File: code/LabTestTools2/src/eda/testtools/resulteditor/ResultEditor.java
package eda.testtools.resulteditor;


import eda.testtools.util.FileUtils;
import java.io.File;
import java.util.List;
import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

/**
 *
 * @author trace
 */
public class ResultEditor extends Application {

    @Override
    public void start(Stage stage) throws Exception {
        FXMLLoader loader = new FXMLLoader(getClass().getResource("FXMLMain.fxml"));
        Parent root = loader.load();
        FXMLMainController cont = loader.getController();

        final Parameters params = getParameters();
        final List<String> parameters = params.getRaw();
        final String filePath = !parameters.isEmpty() ? parameters.get(0) : null;

        if (filePath == null) {
            System.out.println("You must specify a file path!");
            System.exit(0);
        }
        if (!FileUtils.isValidFile(filePath)) {
            System.err.println("Specified file is not valid!");
            System.exit(0);
        }
        cont.setFilePath(filePath);

        Scene scene = new Scene(root);

        stage.setScene(scene);
        stage.show();
    }

    public static void init(String[] args) {
        launch(args);
    }

}

File: code/LabTestTools2/src/eda/testtools/resulteditor/ResultEditor.kt
package eda.testtools.resulteditor

import eda.testtools.util.FileUtils
import eda.tools.main.Constants
import java.nio.file.Files
import java.nio.file.Paths
import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage

/**
 *
 * @author trace
 */
class ResultEditor : Application() {
    override fun start(stage: Stage) {
        val loader = FXMLLoader(this::class.getResource("FXMLMain.fxml"))
        val root = loader!!.load()
        val cont = loader!!.getController()
        val params = getParameters()
        val parameters = params!!.getRaw()
        var filePath = if (parameters.size() > 0 && Files.exists(Paths.get(parameters!!.get(0))))
            parameters!!.get(0)
        else
            "./"
        val inputFile = FileUtils.findFileWithSuffix(filePath,
                Constants.LAB_TESTS_RESULT_SUFFIX)
        if (inputFile == null) {
            System.out.println("Could not find a file to edit!")
            System.exit(0)
        }
        filePath = inputFile!!.getPath()
        System.out.println("Using $filePath as input file.")
        cont!!.setFilePath(filePath)
        val scene = Scene(root)
        stage.setScene(scene)
        stage.show()
    }

    companion object {
        fun init(args: Array<String>?) {
            launch(args)
        }
    }
}